src/main/java/org/apache/sysds/common/Types.java

```diff
@@ -76,7 +76,7 @@ public boolean isUnknown() {
 	 */
 	public enum ValueType {
 		UINT4, UINT8, // Used for parsing in UINT values from numpy.
-		FP32, FP64, INT32, INT64, BOOLEAN, STRING, UNKNOWN,
+		FP32, FP64, INT32, INT64, BOOLEAN, TRUE_BOOLEAN, STRING, UNKNOWN,
```

Comment:
Is it possible to reuse the existing BOOLEAN value type instead of adding a new one?

Reply:
I can reuse the old BOOLEAN type, yes. The reason for having two boolean types right now is to compare the bitset boolean denseblocks and the "true" boolean array denseblocks.

I would suggest renaming the "old" BOOLEAN, which currently is the bitset implementation, to BITSET and using BOOLEAN for the boolean array. That naming would make things more clear.
What do you think?

Reply:
I agree. You can rename those to BITSET and BOOLEAN.

Reply:
I have to revert this renaming. This Valuetype is used for so much more than just denseblocks, which I did not get the first time...

I will reuse the existing Valuetype.BOOLEAN and use another mechanism to create BooleanArrays when desired.
In the end, it makes a lot more sense when both DenseBlockBool implementations use the same ValueType, since both store booleans in the end.

Reply:
Yes. That is a better approach and makes integrating the boolean type into the system easier.